src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -5565,7 +5565,12 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             ASR::asr_t* assign = ASR::make_Assignment_t(al, x.base.base.loc,
                             ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, tmp_assign_variable_sym)),
                             target, nullptr);
-            current_body->push_back(al, ASRUtils::STMT(assign));
+            if(current_body){
+                current_body->push_back(al, ASRUtils::STMT(assign));
+            }else{
+                global_init.reserve(al, x.n_body + 1);
+                global_init.push_back(al,assign);
+            }
```

Comment:
```suggestion
            if (current_body) {
                current_body->push_back(al, ASRUtils::STMT(assign));
            } else {
                global_init.reserve(al, 1);
                global_init.push_back(al, assign);
            }
```

We need to reserve only 1 unit for the new assignment. `x.n_body` is the total number of statements present inside the `for` loop body.

Reply:
Actually the reserve call isn't even needed.

Reply:
@tanay-man Please add the whitespaces at the appropriate places.